pkg/scalers/kafka_scaler.go

```diff
@@ -291,6 +292,10 @@ func parseKerberosParams(config *scalersconfig.ScalerConfig, meta *kafkaMetadata
 	}
 	meta.kerberosConfigPath = path
 
+	if config.AuthParams["serviceName"] != "" {
```

Comment:
Do you think its better to rename `serviceName` to `gssapiServiceName` or something similar, just to make the field clearer as we have quite a number of authentication method supported ?

Reply:
Agree, see: https://github.com/kedacore/keda-docs/pull/1305#discussion_r1486130623

Reply:
@JorTurFer and @dttung2905 - Can we not put the other scaler as an obstacle to update this one. We are already seeing other differences between the two and we need to move forward with the changes on this one. I am not sure why we would wait if these changes are proven to be good changes and a current gap in the configuration. I would ask for someone else in the community who is interested in that scaler to take it on. Or we might be able to contribute further at a later time but this scaler is on our critical path for an effort.

A reasonable ask to proceed? I am fine with the suggested rename. Just taking on the whole other scaler as a prerequisite for a good change to this one seems a bit overkill.

Thank you for your assistance and oversight on this powerful project!

Reply:
Hello @hayscoder ,
Maybe I've understood the message wrongly, but I think that your last comment is a bit disrespectfully. I totally understand that this can be blocking you and your scenario, but we are looking for the best to the project and not just to your scenario. 

> We are already seeing other differences between the two and we need to move forward with the changes on this one. I am not sure why we would wait if these changes are proven to be good changes and a current gap in the configuration

Literally nothing blocks you from forking the current PR, build it by yourself and go ahead with the current changes.

> I would ask for someone else in the community who is interested in that scaler to take it on. Or we might be able to contribute further at a later time but this scaler is on our critical path for an effort.

As KEDA members, we try to do our best effort maintaining the project for free, but I totally understand that you have business goals in mind. We don't provide any kind of enterprise support, you can check in [our website](https://keda.sh/support/#commercial-support) some vendors which I can assume that will be glad offering the custom features that you need (after signing a contract with them).

As I said, you can just for KEDA and include all the custom features that you want, not needed us or our approval for them. If maintaining your own branch with custom features isn't something feasible in your business case, you can extend current KEDA supported cases by creating your own server as you wish, and connecting KEDA to it through [Metrics API Scaler](https://keda.sh/docs/2.13/scalers/metrics-api/) or [External Scaler](https://keda.sh/docs/2.13/scalers/external/). This will allow you to control all the changes that you want, just coding the scaler part and not the whole scaling features.

The last but not least, we gave feedback about the naming 3 weeks ago, and you have answered now saying that this is required for your critical path, 3 weeks later.

Reply:
Truly sorry you felt any disrespect. I did not intend any at all. I was more trying to see some hope of a shorter path towards success. Maybe it was some of my desperation coming out in a "is this the only way?" type of way. I will try to be more aware of how it could come across in a public forum without knowing each other or the background. As you noted accurately, it has been 3 weeks. In those 3 weeks, we have been looking at various obstacles and options with security teams and redhat support. Lots of cooks in the kitchen and we have been spending most of our time overcoming some more immediate obstacles (not related to this PR) to using Keda with the Kafka scaler. 

Thank you as well for taking the time in listing out some of the alternatives. I really do appreciate the effort. We have been looking at some of those as well.

You are correct that I should have worked with Arun to do the naming change sooner even though we were working with those other teams in the background. My miss and I dropped that amongst spinning my other plates. I apologize for that miss.

Thank you again for the time that you put forth towards this powerful project.

Reply:
No worries, let's try to push this feature to be merged as soon as possible 😄 

As I said [here](https://github.com/kedacore/keda/pull/5483#issuecomment-1973387704), during the last community meeting we decided to deprecate the other scaler, so we just need to add the new parameter to [this test collection](https://github.com/kedacore/keda/blob/main/pkg/scalers/kafka_scaler_test.go#L135), modifying a bit [this test](https://github.com/kedacore/keda/blob/main/pkg/scalers/kafka_scaler_test.go#L375) to validate the new field and that could be all the pending tasks

Reply:
👍